subcommittee/coding-guidelines/meetings/2024-December-04/minutes.md

```diff
@@ -28,14 +48,88 @@ Supplemental material to the agenda can be found on the [GitHub repo](https://gi
+* PL: The idea would be to target both audiences that use standard library and not
+* CP: [Embassy](https://github.com/embassy-rs/embassy) would be a good example of unsafe usage
+* AF: There are many patterns in the C world that are accepted, and giving people design patterns would be useful to help the transition
+* JM: In response to the previous Practicum group proposal, agreed good first steps.
+* PL: Proposal for document review task group, to review the [Rust unsafe code
+  guidelines](https://github.com/rust-lang/unsafe-code-guidelines)
+* DD: Agreed, would be beneficial
+* It was suggested that multiple people should review the guidelines
+
+### Practicum Chapter Task Force Progress
+
+* JM: Mostly looking where to start, and perhaps annotating the examples as suggested earlier
+* JM: Would the idea to be supplement the Learn Unsafe Rust book with coding guidelines
+* PL: Spoke to an author the book and was open to contributions and may even be able to transfer ownership to the Rust Foundation
+* JJ: The book has a good starting framework
+
+### Docs Review Task Force Progress
+
+* PL: Mostly review the unsafe code guidelines glossary
+
+### Round Table
+
+* WC: Interested in supply chain security of rust packages?
+* JM: This may be more relevant to the Tooling committee
+* AF: This may be out of scope, but is certainly would be looked at from a regulatory perspective
+* DD: The largest extent that of the coding guidelines would be tell users to have the coding guidelines apply to used packages as well
+* WP: Google has [Rust crate audits](https://github.com/google/rust-crate-audits), using [Cargo Vet](https://github.com/mozilla/cargo-vet)
+
+---
+
+* OB: Wanted to determine what unsafe/undefined behaviour is in reference to the hardware
+* PL: There is a section on undefined behaviour in the Learn Unsafe Rust book
+* OB: Can share document on safety in the scope of low level driver
+
+---
+
+* AH: Is the meeting going to be in January?
+* JM: It will be nearby to Rust Nation in February, on the 19th
+* PL: Would the venue be outside of Rust Nation?
+* JM: **It would be outside of the venue, and would not require tickets to Rust Nation**
+* JM: About 20 people have responded to the poll as going face to face
+
+---
+
+* JP: Different levels of assurance for different guidelines may be useful
+* AF: The definition of unsafe, means that the code will be unproven
+* JP: It should be clear in the guidelines that some unsafe should be used
+* JM: We should have specific recommendations with what to do with unsafe code
+* JJ: We shouldn’t just focus on the unsafe keyword
+* AF: Deadlocks and checked/unchecked arithmetic are examples of stuff outside of this
+* WP: Would this committee want to bless Cargo Vet or create recommendations for crates?
+* PL: Getting involved with Cargo Vet is a good idea
+* PL: There is a proposal to change the unsafe keyword
```

Comment:
In particular, there is a [thought](https://github.com/rust-lang/rfcs/pull/3484#issuecomment-2039501484) Niko had to try to distinguish when code should be marked as `trusted`, i.e. I, the programmer, have verified that although some guardrails are removed in the below section of code, we keep the car on the road.

Seems like a [Rust 2027 consideration](https://github.com/rust-lang/rfcs/pull/3484#issuecomment-2039501484) at this point.

You could include a little of this detail in something of an addendum with the links. What do you think?

Reply:
I agree, I vaguely remember you mentioning more detail about that during the meeting that I couldn't capture in time. I have added the link and reformatted the statement here.